tests/manage/z_cluster/cluster_expansion/test_add_capacity.py

```diff
@@ -17,35 +12,10 @@ class TestAddCapacity(ManageTest):
     """
     Automates adding variable capacity to the cluster while IOs running
     """
-    @pytest.mark.parametrize(
-        argnames=[
-            "node_multiplier", "capacity"
-        ],
-        argvalues=[
-            pytest.param(
-                *[3, '2000Gi'],
-            ),
-        ]
-    )
-    def test_add_capacity(self, node_multiplier, capacity):
+    def test_add_capacity(self):
         """
         Test to add variable capacity to the OSD cluster while IOs running
-
-        Args:
-        node_multiplier: the number of OSD to add per worker node
-        capacity: the storage capacity of each OSD
         """
-        dt = config.ENV_DATA['deployment_type']
-        if dt == 'ipi':
-            storage_cluster = machine_utils.get_storage_cluster(namespace=defaults.ROOK_CLUSTER_NAMESPACE)
-            worker_nodes = len(get_typed_nodes())
-            machine_utils.add_capacity(storagecluster_name=storage_cluster, count=worker_nodes * node_multiplier)
-            machine_utils.add_storage_capacity(storagecluster_name=storage_cluster, capacity=capacity)
-            pod_obj = ocp.OCP(kind=constants.POD, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
-            assert pod_obj.wait_for_resource(
-                condition=constants.STATUS_RUNNING, selector=constants.OSD_APP_LABEL,
-                resource_count=worker_nodes * node_multiplier, timeout=600
-            ), "OSD pods failed to reach RUNNING state"
-        else:
-            pytest.skip("UPI not yet supported")
-        # ToDo run IOs
+        osd_size = storage_cluster.get_osd_size()
+        result = storage_cluster.add_capacity(osd_size)
+        assert result, logger.info("Test Failed, new pods failed reaching running state")
```

Comment:
Post successful addition of capacity it would be better to check for the ceph health state and confirm that it is Healthy (we will have to to wait till the rebalance completes if any).
through which,
1) we can confirm that there are no issues wrt ceph post addition of add-capacity
2) If the number of OSDs added are more, rebalance will take more time. So, there are high chances that the next running test might fail since ceph health status is not healthy.
Thoughts?

Reply:
Thanks Prasad. test_add_capacity.py will have more changes in coming days like entry and exit criteria verification. The main focus of this PR for now is to implement the core part of cluster expansion, the 'add_capacity' function. 

Shreekara/Anna, Request you to move the code "pod.ocp.wait_for_resource" out of the 'add_capacity' function. We need add_capacity function to just 'patch' and come out. Rest of the things like verification should be handled by the test script calling it. This way it will help both tier1 and tier4 tests to use this function w/o modifying. For tier4 we need to introduce the failures while rebuild or osd addition is going on. If we make add_capacity function to wait for the above to complete in it then it will be difficult for tier4 tests introduce the failure at the right time.